File: base64_and_file/src/main/java/com/phone/base64_and_file/model/Base64AndFileModelImpl.java
package com.phone.base64_and_file.model;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;

import com.phone.base64_and_file.bean.Base64AndFileBean;
import com.phone.base64_and_file.thread_pool.Base64ToPictureThreadPool;
import com.phone.base64_and_file.thread_pool.CompressedPictureThreadPool;
import com.phone.base64_and_file.thread_pool.PictureToBase64ThreadPool;
import com.phone.common_library.callback.OnCommonSingleParamCallback;
import com.phone.common_library.manager.LogManager;

public class Base64AndFileModelImpl implements IBase64AndFileModel {

    private static final String TAG = Base64AndFileModelImpl.class.getSimpleName();
    private CompressedPictureThreadPool compressedPictureThreadPool;
    private PictureToBase64ThreadPool pictureToBase64ThreadPool;
    private Base64ToPictureThreadPool base64ToPictureThreadPool;

    private Handler handler;

    public Base64AndFileModelImpl() {
        handler = new Handler(Looper.getMainLooper());
    }

    @Override
    public void showCompressedPicture(Context context,
                                      Base64AndFileBean base64AndFileBean,
                                      OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback) {
        LogManager.i(TAG, "showCompressedPicture");

        compressedPictureThreadPool = new CompressedPictureThreadPool(
                context,
                base64AndFileBean);
        compressedPictureThreadPool.setOnCommonSingleParamCallback(new OnCommonSingleParamCallback<Base64AndFileBean>() {
            @Override
            public void onSuccess(Base64AndFileBean success) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onSuccess(success);
                    }
                });
            }

            @Override
            public void onError(String error) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onError(error);
                    }
                });
            }
        });
        compressedPictureThreadPool.submit();
    }

    @Override
    public void showPictureToBase64(Base64AndFileBean base64AndFileBean,
                                    OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback) {
        LogManager.i(TAG, "showPictureToBase64");

        pictureToBase64ThreadPool = new PictureToBase64ThreadPool(base64AndFileBean);
        pictureToBase64ThreadPool.setOnCommonSingleParamCallback(new OnCommonSingleParamCallback<Base64AndFileBean>() {
            @Override
            public void onSuccess(Base64AndFileBean success) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onSuccess(success);
                    }
                });
            }

            @Override
            public void onError(String error) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onError(error);
                    }
                });
            }
        });
        pictureToBase64ThreadPool.submit();
    }

    @Override
    public void showBase64ToPicture(Base64AndFileBean base64AndFileBean,
                                    OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback) {
        LogManager.i(TAG, "showBase64ToPicture");

        base64ToPictureThreadPool = new Base64ToPictureThreadPool(base64AndFileBean);
        base64ToPictureThreadPool.setOnCommonSingleParamCallback(new OnCommonSingleParamCallback<Base64AndFileBean>() {
            @Override
            public void onSuccess(Base64AndFileBean success) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onSuccess(success);
                    }
                });
            }

            @Override
            public void onError(String error) {
                handler.post(new Runnable() {
                    @Override
                    public void run() {
                        onCommonSingleParamCallback.onError(error);
                    }
                });
            }
        });
        base64ToPictureThreadPool.submit();
    }
}

File: library_base64_and_file/src/main/java/com/phone/library_base64_and_file/model/Base64AndFileModelImpl.kt
package com.phone.library_base64_and_file.model

import android.content.Context
import android.os.Handler
import android.os.Looper
import com.phone.library_base64_and_file.bean.Base64AndFileBean
import com.phone.library_base64_and_file.thread_pool.Base64ToPictureThreadPool
import com.phone.library_base64_and_file.thread_pool.CompressedPictureThreadPool
import com.phone.library_base64_and_file.thread_pool.PictureToBase64ThreadPool
import com.phone.library_base.callback.OnCommonSingleParamCallback
import com.phone.library_base.manager.LogManager.i

class Base64AndFileModelImpl : IBase64AndFileModel {

    private val TAG = Base64AndFileModelImpl::class.java.simpleName
    private var handler: Handler

    init {
        handler = Handler(Looper.getMainLooper())
    }

    override fun showCompressedPicture(
        context: Context,
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    ) {
        i(TAG, "showCompressedPicture")
        val compressedPictureThreadPool = CompressedPictureThreadPool(
            context,
            base64AndFileBean
        )
        compressedPictureThreadPool.setOnCommonSingleParamCallback(object :
            OnCommonSingleParamCallback<Base64AndFileBean?> {
            override fun onSuccess(success: Base64AndFileBean?) {
                handler.post { onCommonSingleParamCallback.onSuccess(success) }
            }

            override fun onError(error: String) {
                handler.post { onCommonSingleParamCallback.onError(error) }
            }
        })
        compressedPictureThreadPool.submit()
    }

    override fun showPictureToBase64(
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    ) {
        i(TAG, "showPictureToBase64")
        val pictureToBase64ThreadPool = PictureToBase64ThreadPool(base64AndFileBean)
        pictureToBase64ThreadPool.setOnCommonSingleParamCallback(object :
            OnCommonSingleParamCallback<Base64AndFileBean?> {
            override fun onSuccess(success: Base64AndFileBean?) {
                handler.post { onCommonSingleParamCallback.onSuccess(success) }
            }

            override fun onError(error: String) {
                handler.post { onCommonSingleParamCallback.onError(error) }
            }
        })
        pictureToBase64ThreadPool.submit()
    }

    override fun showBase64ToPicture(
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    ) {
        i(TAG, "showBase64ToPicture")
        val base64ToPictureThreadPool = Base64ToPictureThreadPool(base64AndFileBean)
        base64ToPictureThreadPool.setOnCommonSingleParamCallback(object :
            OnCommonSingleParamCallback<Base64AndFileBean?> {
            override fun onSuccess(success: Base64AndFileBean?) {
                handler.post { onCommonSingleParamCallback.onSuccess(success) }
            }

            override fun onError(error: String) {
                handler.post { onCommonSingleParamCallback.onError(error) }
            }
        })
        base64ToPictureThreadPool.submit()
    }

    override fun detachViewModel() {
        handler.removeCallbacksAndMessages(null)
    }
}